pdns/ueberbackend.cc

```diff
@@ -120,12 +121,18 @@ bool UeberBackend::getDomainInfo(const DNSName &domain, DomainInfo &di, bool get
 
 bool UeberBackend::createDomain(const DNSName &domain, const DomainInfo::DomainKind kind, const vector<ComboAddress> &masters, const string &account)
 {
+  bool success = false;
+  int zoneId;
   for(DNSBackend* mydb :  backends) {
-    if(mydb->createDomain(domain, kind, masters, account)) {
-      return true;
+    if(mydb->createDomain(domain, kind, masters, account, &zoneId)) {
+      success = true;
+      break;
     }
   }
-  return false;
+  if (success) {
+    g_domainCache.add(domain, zoneId);  // make new domain visible
+  }
```

Comment:
For the peanut gallery: Code scanning isn't smart enough to know that `zoneId` is guarded by `success` and success will only be true if zoneId is filled in.

To make things more complicated, `zoneId` is a pointer not a reference on the callee side, and callees obviously only write to it if the pointer is non null.